File: 20. Valid Parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        brac_map = {')':'(', '}':'{', ']':'['}
        res = []
        for i in s:
            if i in brac_map.values():
                res.append(i)
            else:
                if not res or res[-1] != brac_map[i]:
                    return False
                res.pop()
        if (len(res) > 0):
            return False
        else:
            return True

File: 20. Valid Parentheses.go
func isValid(s string) bool {
    brac_map := map[rune]rune{'}': '{', ']': '[', ')': '('}
    var brac_track []rune

    for _,char := range s {
        if _, ok := brac_map[char]; !ok {
            brac_track = append(brac_track, char)
        } else {
            if len(brac_track) == 0 || brac_track[len(brac_track)-1] != brac_map[char] {
                return false
            } 
            brac_track = brac_track[:len(brac_track)-1]
        }
    }
    fmt.Println(brac_track)

    return len(brac_track) == 0
    
}